contracts/contracts/MessageProcessor.sol

```diff
@@ -82,51 +84,36 @@ contract MessageProcessor is Ownable, SnarkCommon, Hasher, CommonUtilities, IMes
       revert NoMoreMessages();
     }
 
-    // The state AccQueue must be merged
-    if (!poll.stateMerged()) {
-      revert StateNotMerged();
-    }
-
     // Retrieve stored vals
-    (, uint8 messageTreeSubDepth, uint8 messageTreeDepth, uint8 voteOptionTreeDepth) = poll.treeDepths();
+    (, uint8 voteOptionTreeDepth) = poll.treeDepths();
     // calculate the message batch size from the message tree subdepth
-    uint256 messageBatchSize = TREE_ARITY ** messageTreeSubDepth;
-
-    (, AccQueue messageAq) = poll.extContracts();
-
-    // Require that the message queue has been merged
-    uint256 messageRoot = messageAq.getMainRoot(messageTreeDepth);
-    if (messageRoot == 0) {
-      revert MessageAqNotMerged();
-    }
+    uint256 messageBatchSize = poll.batchSizes();
 
     // Copy the state and ballot commitment and set the batch index if this
     // is the first batch to process
     if (numBatchesProcessed == 0) {
       uint256 currentSbCommitment = poll.currentSbCommitment();
       sbCommitment = currentSbCommitment;
-      (, uint256 numMessages) = poll.numSignUpsAndMessages();
-      uint256 r = numMessages % messageBatchSize;
 
-      currentMessageBatchIndex = numMessages;
+      poll.padLastBatch();
+      batchHashes = poll.getBatchHashes();
+      currentBatchIndex = batchHashes.length;
```

Comment:
this could be optimized:

```
      uint256[] memory _batchHashes = poll.getBatchHashes();
      batchHashes = _batchHashes;
      currentBatchIndex = _batchHashes.length;
```

Reply:
resolved

Reply:
I don't see the change was it pushed?